### ISSUE TITLE
[HOLD for payment 2023-04-05][$1000] When initiating chat with new user `This is the beginning...` is not displayed in LHN as with group message

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#15799]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Go to the search
2. Search new user after that check on side panel
3. Now check that on group chat
4. Create n new group chat

## Expected Result:
This message should be displayed below the username "You can also use the + button below to send or request money!"

## Actual Result:
Before sending any message to user, email address showing below the username

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.81-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/224175068-d750bbb4-b0e6-4ba4-9339-e7e0caeef5ab.mp4

https://user-images.githubusercontent.com/43996225/224175114-148dee29-8dbb-4b43-adf1-5d1cdeb77fc2.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @ayazhussain79
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1678200365765579

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01fca09f95526ac422</li>
        <li>Upwork Job ID: 1634360065103667200</li>
        <li>Last Price Increase: 2023-03-22</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @kevinksullivan (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
Triggered auto assignment to @AndrewGable (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01fca09f95526ac422

**Comment by MelvinBot** [comment]
Current assignee @kevinksullivan is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @thesahindia (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @dangrous (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

The LHN subtitle of 1:1 chat only shows email address of the user instead of 'This is the beginning of your chat...' which is not the same with a group chat.

### What is the root cause of that problem?

We only show 'This is the beginning of your chat...' on a chat with multiple participants.

### What changes do you think we should make in order to solve the problem?


I think this is expected. Looking at the code, only chat with multiple participants that has the text 'This is the beginning of your chat history with {user1}, {user2}'. As you can see, it lists all user in that group (even though we can't see it), but in a 1:1 chat, there is no other people beside that user itself. So, I'm guessing that's why only for group we have the text 'This is the beginning of your chat history with {user1}, {user2}'. If we still want to show 'This is the beginning of your chat...' for 1:1 chat, we can simply remove the multiple participants validation here.

https://github.com/Expensify/App/blob/b69b5f3f31d3054dd9eadb1a284d5efb490daf15/src/libs/SidebarUtils.js#L270-L273

**Comment by Taimoor0217** [comment]
# Proposal
## Please re-state the problem that we are trying to solve in this issue.
The subtitle below the LHN shows the email address of the user instead of the desired message.

## What is the root cause of that problem?
The root problem is with the incorrect or no assignment of the variable `lastMessageText` in the case of 1:1 chats. As you can see from the following line.
https://github.com/Expensify/App/blob/b69b5f3f31d3054dd9eadb1a284d5efb490daf15/src/libs/SidebarUtils.js#L283
We default the alternate text to `personalDetail.login` when the `lastNessageText` variable is not defined. And this variable stays undefined or empty because we only set the value of the variable in the following cases.

### It is an archived room or policy expense chat.
https://github.com/Expensify/App/blob/b69b5f3f31d3054dd9eadb1a284d5efb490daf15/src/libs/SidebarUtils.js#L258

### The chat has multiple users
https://github.com/Expensify/App/blob/b69b5f3f31d3054dd9eadb1a284d5efb490daf15/src/libs/SidebarUtils.js#L270

## What changes do you think we should make in order to solve the problem?
### ATTENTION: Probable Typo in the issue's expected results section.
I think there is a typo in the expected results section. The title says that we should display the 'This is the beginning text....', however, the expected results say that we should show  "You can also use the + button below to send or request money!" as a subtitle. Hence I am going to propose a solution for both cases.

## Setting the subtitle to `This is the beginning ....` message.
In this case, we can remove the condition of multiple users from assignments in the following if condition. This way we will let our single chats to qualify to have the same subtitle.
https://github.com/Expensify/App/blob/b69b5f3f31d3054dd9eadb1a284d5efb490daf15/src/libs/SidebarUtils.js#L270

## Setting the subtitle to  `"You can also use the + button below to send or request money!"`
In this case, we can add the `personalDetail.login` with our message in the following line.
https://github.com/Expensify/App/blob/b69b5f3f31d3054dd9eadb1a284d5efb490daf15/src/libs/SidebarUtils.js#L283

**Comment by MelvinBot** [comment]
📣 @Taimoor0217! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by Taimoor0217** [comment]
Contributor details
Your Expensify account email: taimoor0217@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01093e6d7144f50f6b

**Comment by MelvinBot** [comment]
✅ Contributor details stored successfully. Thank you for contributing to Expensify!

**Comment by ayazhussain79** [comment]
**Proposal**
I identified that issue
**Please re-state the problem that we are trying to solve in this issue.**
On LHN displayed email address/username of the user instead of last App message 

**What is the root cause of that problem?**
we are not showing last message "This is the beginning of your chat.." or "you can also use + button" on 1:1 chat 

**What changes do you think we should make in order to solve the problem?**

we are not showing this message on 1:1 chat 'This is the beginning of your chat..' on 1:1 chat.
but on group chat we are showing this message,
so we want to show 'This is the beginning of your chat..' this message then we need to remove hasMultipleParticipants from the if condition 
her is the code:
Line 269: if (!lastMessageText)//here i removed hasMultipleParticipants from the condition
 {
         ---
        result.alternateText = lastMessageText || Str.removeSMSDomain(personalDetail.login);
}
Result:
![Screenshot_86](https://user-images.githubusercontent.com/56740856/224482380-cf8b231f-93e9-4908-8c6e-27faadc7e035.png)

but if we want to show this message the 'you can also use + button' then here is the code:
Line 269:  if (!lastMessageText) {
            // Here we get the beginning of chat history message and append the display name for each user, adding pronouns if there are any.
            // We also add a fullstop after the final name, the word "and" before the final name and commas between all previous names.
            let useBeginning = 'reportActionsView.beginningOfChatHistory';
            let usePlus  = 'reportActionsView.usePlusButton';
            let beginningOrPlus = hasMultipleParticipants ? useBeginning:usePlus;//if hasMultipleParticipants true user 'beginning message' if false user 'Plus Button message'
            lastMessageText = Localize.translate(preferredLocale, beginningOrPlus)//and using beginningOrPlus here 
                + _.map(displayNamesWithTooltips, ({displayName, pronouns}, index) => {
                    const formattedText = _.isEmpty(pronouns) ? displayName : `${displayName} (${pronouns})`;
                    if (index === displayNamesWithTooltips.length - 1) { return `${formattedText}.`; }
                    if (index === displayNamesWithTooltips.length - 2) { return `${formattedText} ${Localize.translate(preferredLocale, 'common.and')}`; }
                    if (index < displayNamesWithTooltips.length - 2) { return `${formattedText},`; }
                }).join(' ');
        }
		////Here I have removed the username from the last message.
        const index = lastMessageText.indexOf('!');
        if (index !== -1) {
            lastMessageText = lastMessageText.substring(0, index + 1);
        }
        result.alternateText = lastMessageText || Str.removeSMSDomain(personalDetail.login);
    }
Result:
![Screenshot_87](https://user-images.githubusercontent.com/56740856/224482424-0470d200-c207-4a7d-95a4-ddb425f1cb11.png)

**Comment by MelvinBot** [comment]
📣 @ayazhussain79! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by ayazhussain79** [comment]
Contributor details
Expensify account email: [hussainayaz79@gmail.com](mailto:hussainayaz79@gmail.com)

Upwork Profile Link: https://www.upwork.com/fl/~018a62c1e70a3fab40

**Comment by dangrous** [comment]
So clarifying - I think we need to figure out the expected behavior here:
1. Why do we differentiate between chats with one person and group chats? I don't really see why we would but I will check in Slack and see if anyone has a good answer. If not, we should definitely consolidate and remove that `hasMultipleParticipants` check.
2. Do we want to show the "This is the beginning..." line (as with group chats) or the "You can also..." line (as in the issue description)? @kavimuru can you confirm where you got that spec?

**Comment by dangrous** [comment]
I'll follow up on Slack tomorrow, haven't had any reaction yet.

**Comment by Taimoor0217** [comment]
@dangrous any updates on this?

Thanks :)

**Comment by dangrous** [comment]
Slowly but surely! Sorry, last week was a bit hectic for a lot of us. Conversation continues... https://expensify.slack.com/archives/C01GTK53T8Q/p1679337013344439?thread_ts=1678735336.127209&cid=C01GTK53T8Q

**Comment by dangrous** [comment]
Okay, answers! Here's what we're going for:

* DM with one person - `This is the beginning of your chat with ...`
* DM with group - `This is the beginning of your chat with …`
* Room - `No activity yet`

We will need new translations for these, which I will put in here when they exist.

We may need new proposals to account for the Room adjustment, since previously we were only dealing with the DM side of things.

**Comment by Taimoor0217** [comment]
# Proposal [Updated with the new Room Requirement]
## Please re-state the problem that we are trying to solve in this issue.
The subtitle below the username  in LHN shows the email address of the user instead of the desired message for 1:1 chats

## What is the root cause of that problem?
The root problem is with the incorrect or no assignment of the variable `lastMessageText` in the case of 1:1 chats. As you can see from the following line.
https://github.com/Expensify/App/blob/b69b5f3f31d3054dd9eadb1a284d5efb490daf15/src/libs/SidebarUtils.js#L283
We default the alternate text to `personalDetail.login` when the `lastNessageText` variable is not defined. And this variable stays undefined or empty because we only set the value of the variable in the following cases.

#### It is an archived room or policy expense chat.
https://github.com/Expensify/App/blob/b69b5f3f31d3054dd9eadb1a284d5efb490daf15/src/libs/SidebarUtils.js#L258

#### The chat has multiple users
https://github.com/Expensify/App/blob/b69b5f3f31d3054dd9eadb1a284d5efb490daf15/src/libs/SidebarUtils.js#L270

## What changes do you think we should make in order to solve the problem?

## 1- Setting the subtitle to `This is the beginning ....` message.
In this case, we can remove the condition of multiple users from assignments in the following if condition. This way we will let our single chats to qualify to have the same subtitle.
https://github.com/Expensify/App/blob/b69b5f3f31d3054dd9eadb1a284d5efb490daf15/src/libs/SidebarUtils.js#L270
#### Proof of Concept
![image](https://user-images.githubusercontent.com/48244930/224462605-ca84734c-a711-45f5-8eb4-56371224473a.png)

## 2- No activity yet in chat rooms.
As we can see in  the follwoing line
https://github.com/Expensify/App/blob/b69b5f3f31d3054dd9eadb1a284d5efb490daf15/src/libs/SidebarUtils.js#L240

The subtitle is created by calling a method called ``getChatRoomSubtitle`

**Comment by bernhardoj** [comment]
Updated [proposal](https://github.com/Expensify/App/issues/15799#issuecomment-1464786096)

**Comment by MelvinBot** [comment]
@dangrous @kevinksullivan @thesahindia this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by MelvinBot** [comment]
📣 @bernhardoj You have been assigned to this job by @dangrous!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by dangrous** [comment]
> Based on the contributing guide, we should edit the original proposal instead.
Oh! Never mind then. I feel like that guidance has changed recently, and I am clearly not up to date. Sorry about that!

**Comment by bernhardoj** [comment]
PR is ready for review.

cc: @thesahindia

**Comment by Taimoor0217** [comment]
@dangrous I guess the approved/merged solution introduces a new issue where the "No activity yet" text is shown for the archived chats, however, the chat unavailable message should be shown. This is introduced because we are completely ignoring the initial subtitle from the `getPolicyName` method as mentioned in my proposal above https://github.com/Expensify/App/issues/15799#issuecomment-1479833574. If the team decides on creating a separate bug for this fix, I would like to take it up. 

Thanks :)

**Comment by bernhardoj** [comment]
> the chat unavailable message should be shown

To be fair, that's not an issue caused by the merged PR. Before the PR, we will just show the workspace name if it's an archived room. There is no chat unavailable message.

**Comment by dangrous** [comment]
Hm I'm not seeing that on dev, though it does seem like I should... 
<img width="369" alt="image" src="https://user-images.githubusercontent.com/5487802/228882998-a8baf05e-760f-4d48-8b38-93fe4b5ff8b7.png">
Do you mind sharing reproduction steps so I can confirm?

If it does show up, I think it's not quite a regression since neither the old or new behavior makes sense haha (this just made it a little more noticeable)... It should presumably show the last chat as normal, or possible a "archived" message? But definitely would be a good bug to fix if it exists and we can create a new issue!

**Comment by Taimoor0217** [comment]
@bernhardoj apologies for the misunderstanding. I might be incorrect, but to my understanding, we still showed the chat unavailable message when there is not activity in a room and it has been archived.

@dangrous I am talking about the scenario, where there is no last message in the chat and then it is archived. I guess in that scenario, it might be better to show a different message than no activity? Maybe the chat unavailable message?

archived room with the last message
![image](https://user-images.githubusercontent.com/48244930/228931999-c05a1823-685e-4fdf-a674-d80b0beed1ac.png)

archived without last message
![image](https://user-images.githubusercontent.com/48244930/228931846-d653b8d8-8e43-479f-a1d8-25defd0fa373.png)

**Comment by dangrous** [comment]
Ohh got it. I think that's fine; it's pretty clear what it means. Thanks for checking on it though!

**Comment by dangrous** [comment]
This didn't get auto posted for some reason, but this was deployed to production on 3/29, so we should handle payment on 4/5 cc @kevinksullivan

We also need to fill out the checklist:

BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

 [@thesahindia / @dangrous] The PR that introduced the bug has been identified. Link to the PR:
 [@thesahindia / @dangrous] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
 [@thesahindia / @dangrous] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
 [@kevinksullivan] Determine if we should create a regression test for this bug.
 [@thesahindia] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md?rgh-link-date=2023-03-31T10%3A44%3A20Z) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.
 [@kevinksullivan] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by kevinksullivan** [comment]
Offers sent to @thesahindia and @bernhardoj !

**Comment by ayazhussain79** [comment]
@kevinksullivan I identified that issue

**Comment by MelvinBot** [comment]
Current assignee @kevinksullivan is eligible for the NewFeature assigner, not assigning anyone new.

**Comment by dangrous** [comment]
Looks good!

**Comment by kevinksullivan** [comment]
Paid + bonus to @bernhardoj .

@thesahindia let me know when you've accepted the new job I sent over! Sorry about the last one.

@ayazhussain79 sent an offer for reporting. Let me know when you accept!

**Comment by bernhardoj** [comment]
@kevinksullivan I haven't receive the bonus, yet.

**Comment by ayazhussain79** [comment]
@kevinksullivan offer accepted

**Comment by kevinksullivan** [comment]
@bernhardoj hm it says paid in upwork? You still didn't receive it?

<img width="619" alt="image" src="https://user-images.githubusercontent.com/22508290/230458991-8bfec811-4bab-454a-8f3d-2132fb94c86a.png">

**Comment by ayazhussain79** [comment]
Thank you

**Comment by bernhardoj** [comment]
> Paid + bonus to @bernhardoj

@kevinksullivan Yes, I have received the initial payment. I thought we are eligible for the 50% bonus based on your previous comment, but I just checked that the PR is merged within 4 days 😆.

**Comment by bernhardoj** [comment]
Ah, I forget about that. Thanks for correcting it!

**Comment by kevinksullivan** [comment]
Ah, sorry for the confusion @bernhardoj . I just sent you a bonus and paid @thesahindia , so this is all set!
